dns/route53/dns.go

```diff
@@ -37,33 +37,50 @@ func NewSRVManager(client *Client, hostedZoneID string, recordName string, ttl u
 // edit provides both add and removal capabilities
 func (s *SRVManager) edit(add bool, srv *net.SRV) error {
 	log.Infof("Looking up SRV resource record set for %s", s.recordName)
-	resourceRecordSet, err := s.client.GetResourceRecordSetByName(s.hostedZoneID, s.recordName, "SRV")
+	currentResourceRecordSet, err := s.client.GetResourceRecordSetByName(s.hostedZoneID, s.recordName, "SRV")
 	if err != nil {
 		return err
 	}
 
-	if resourceRecordSet == nil {
+	var resourceRecordSet *route53Client.ResourceRecordSet
+
+	if currentResourceRecordSet == nil {
```

Comment:
We should move this further down - we only need to bother creating the record if we're actually adding something.

Reply:
Are you referring to `LOC:45`? or the condition below?

Reply:
I'm referring to the log line immediately below, and the `&route53Client.ResourceRecordSet{` initialization. This logs it'd create a new record even when the `edit` is essentially called during a remove operation (Imagine starting `srv-announcer` on a not-yet-existing record, while the healthcheck isn't yet met).

Given we only care about `TTL`, `Name` and `Type` during creation, what about only working on a `var resourceRecords []*ResourceRecord`, which is either filled from `.ResourceRecords` of what `s.client.GetResourceRecordSetByName`, or an empty slice (if it was`nil`).

Then we could delay the whole `&route53Client.ResourceRecordSet` initialization (and log line) to the button, and only do it if we really determined we need to create the record at all.

This should in more readable code, and less misleading log messages.